### ISSUE TITLE
[FEATURE] Add side menu peeking when using gesture navigation

### ISSUE THREAD

**Issue by sinfullad** [BrightDV/BoxBox#26]
**Is your feature request related to a problem? Please describe.**
We should able to swipe open the side menu without needing to disable gesture or tap the hamburger menu

**Describe the solution you'd like**
Add peeking/swipe at 45° degree to open the side menu, like Infinity for Reddit has.
https://9to5google.com/2019/09/08/android-10-side-hamburger-menus/

**Additional context**
https://user-images.githubusercontent.com/67637299/203293036-76c26ac1-4740-4b15-b3e6-2d2d144880c9.mp4

**Comment by BrightDV** [comment]
It should be fixed by 5c8e90e. You can try it with the apk of c9adc0b (https://github.com/BrightDV/BoxBox/suites/9524675517/artifacts/452477597)

**Comment by sinfullad** [comment]
> It should be fixed by 5c8e90e. You can try it with the apk of c9adc0b (https://github.com/BrightDV/BoxBox/suites/9524675517/artifacts/452477597)

The link doesn't seem to work?

**Comment by BrightDV** [comment]
Strange... go [here](https://github.com/BrightDV/BoxBox/actions/runs/3558306675), scroll down to ` Artifacts` and click on ` app-release.apk`

**Comment by sinfullad** [comment]
> Strange... go [here](https://github.com/BrightDV/BoxBox/actions/runs/3558306675), scroll down to ` Artifacts` and then click on ` app-release.apk`. Also, you must be logged in, it is a limitation on GitHub's side.

Ok, it works now. I downloaded the latest commit you did, and the swipe works from anywhere now. I got confused at first, expecting it to work like Infinity lol

Also, the one where you fixed the unsupported Atom widget blanked the page for me? Probably needs more time

![Screenshot_20221128_023607.jpg](https://user-images.githubusercontent.com/67637299/204156038-0b878234-c2f2-4d1f-a438-16a5e51f44ad.jpg)

**Comment by BrightDV** [comment]
> > Strange... go [here](https://github.com/BrightDV/BoxBox/actions/runs/3558306675), scroll down to `Artifacts` and then click on `app-release.apk`. Also, you must be logged in, it is a limitation on GitHub's side.
> 
> Ok, it works now. I downloaded the latest commit you did, and the swipe works from anywhere now. I got confused at first, expecting it to work like Infinity lol

Yes, it is more temporary but at least it doesn't work too badly...

> Also, the one where you fixed the unsupported Atom widget blanked the page for me? Probably needs more time

I fixed it again in faa70bc, a variable type problem...

**Comment by sinfullad** [comment]
> > > Strange... go [here](https://github.com/BrightDV/BoxBox/actions/runs/3558306675), scroll down to `Artifacts` and then click on `app-release.apk`. Also, you must be logged in, it is a limitation on GitHub's side.
> > 
> > Ok, it works now. I downloaded the latest commit you did, and the swipe works from anywhere now. I got confused at first, expecting it to work like Infinity lol
> 
> Yes, it is more temporary but at least it doesn't work too badly...
> 
> > Also, the one where you fixed the unsupported Atom widget blanked the page for me? Probably needs more time
> 
> I fixed it again in faa70bc, a variable type problem...

Don't worry, your temporary solution is less of a hassle than the official one, especially when I'm using something One Hand Operation+ from Good Lock

Also, the widget is fixed now, but some driver names got buried
![Screenshot_20221130_062048.jpg](https://user-images.githubusercontent.com/67637299/204670350-3d3faf1f-eebf-49e6-aa0a-f6407e7c6246.jpg)

**Comment by sinfullad** [comment]
> Can you share the link of the article?

It's the same article as above.

Also a good time to suggest adding a search function to the app's toolbar, so we don't have scroll through the news as often

https://www.formula1.com/en/latest/article.2023-f1-grid-all-the-drivers-and-teams-racing-next-season.39kWUtoj6oL2HOMcLGO70V.html

**Comment by BrightDV** [comment]
> Also a good time to suggest adding a search function to the app's toolbar, so we don't have scroll through the news as often

It is a good idea, but I can't because it is not implemented on the server side (you can't search on the website). Maybe I can add a button to save/like an article for an easier access.

**Comment by sinfullad** [comment]
> > Also a good time to suggest adding a search function to the app's toolbar, so we don't have scroll through the news as often
> 
> It is a good idea, but I can't because it is not implemented on the server side (you can't search on the website). Maybe I can add a button to save/like an article for an easier access.

While you are at it, how about adding History too?

**Comment by BrightDV** [comment]
> > > Also a good time to suggest adding a search function to the app's toolbar, so we don't have scroll through the news as often
> > 
> > 
> > It is a good idea, but I can't because it is not implemented on the server side (you can't search on the website). Maybe I can add a button to save/like an article for an easier access.
> 
> While you are at it, how about adding History too?

Added! Also, I'm closing this as the issue is resolved.

**Comment by BrightDV** [comment]
I added a search functionality using SearXNG and 5 instances to search for articles.
However, the requests are often blocked because of rate limits.

**Comment by sinfullad** [comment]
Does the search functionality works just like a normal SearXNG would? Hiw do you filter out other things while searching for F1 articles?

I never tried RSS before, or whether F1's official allows RSS but I'm wondering whether you can use RSS as a way to store and search articles?

**Comment by BrightDV** [comment]
I created an issue for that, see [my comment](https://github.com/BrightDV/BoxBox/issues/41#issuecomment-1356770268).
